src/internal/nozzle/nozzle.go

```diff
@@ -328,6 +355,12 @@ func (n *Nozzle) createPointsFromGauge(envelope *loggregator_v2.Envelope) []*rpc
 			"source_id": envelope.GetSourceId(),
 			"unit":      metric.GetUnit(),
 		}
+
```

Comment:
Here envelope.GetTags() is called several times, it would be good to get tags outside of for loop and reuse in other places as well 
Also the check hasMatchedTags(tags) should be moved outside of loop as well to check once for onvelope with multiple metrics.
If it is possibel you could consider other improvements on existing code by moving static parts outside of for loop.